xclim/indices/_agro.py

```diff
@@ -1418,3 +1419,53 @@ def effective_growing_degree_days(
     egdd = aggregate_between_dates(deg_days, start=start, end=end, freq=freq)
 
     return to_agg_units(egdd, tas, op="delta_prod")
+
+
+@declare_units(tasmin="[temperature]")
+def hardiness_zones(
+    tasmin: xarray.DataArray, window: int = 30, method: str = "usda", freq: str = "YS"
+):
+    """Hardiness zones.
+
+    Hardiness zones are a categorization of the annual extreme temperature minima, averaged over a certain period.
+    The USDA method defines 14 zones, each divided into two sub-zones, using steps of 5°F, starting at -60°F.
+    The Australian National Botanic Gardens method defines 7 zones, using steps of 5°C, starting at -15°C.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+        Minimum temperature.
+    window : int
+        The length of the averaging window, in years.
+    method : {'usda', 'anbg'}
+        Whether to return the American (`usda`) or the Australian (`anbg`) classification zones.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        {method} hardiness zones.
+        US sub-zones are denoted by using a half step. For example, Zone 4b is given as 4.5.
+        Values are given at the end of the averaging window.
+
+    References
+    ----------
+    :cite:cts:`usda_2012,dawson_plant_1991`
+    """
+    if method.lower() == "usda":
+        zone_min, zone_max, zone_step = "-60 degF", "70 degF", "5 degF"
+
+    elif method.lower() == "anbg":
+        zone_min, zone_max, zone_step = "-15 degC", "20 degC", "5 degC"
+
+    else:
+        raise NotImplementedError(
+            f"Method must be one of `usda` or `anbg`. Got {method}."
+        )
+
+    tn_min = tasmin.resample(time=freq).min().rolling(time=window).mean()
```

Comment:
Do we not want to take min only in the time dime `min(dim='time')`?

Reply:
We do.

Reply:
In the end this is exactly `xc.indices.tn_min(time=freq).rolling(time=window).mean()`

Reply:
Essentially, haha

Reply:
I think xarray's behavior might have changed... `.min()` seems to only average across the resampled dim as wanted.  I thought you were going to end up averaging over all dims ... been a while since I coded an xclim index I guees